packages/lib-sourcify/src/lib/solidityCompiler.ts

```diff
@@ -140,6 +170,51 @@ async function fetchSolcFromGitHub(
   return false;
 }
 
+async function validateSolcJsPath(soljsonPath: string): Promise<boolean> {
+  try {
+    const solcjsImports = await import(soljsonPath);
+    const soljson = solc.setupMethods(solcjsImports);
+    if (soljson) {
+      return true;
+    }
+    return false;
+  } catch (e) {
+    return false;
+  }
+}
+
+async function fetchSolcJsFromGitHub(
```

Comment:
Do we really need a whole new function for this? It seems to be overlapping a lot with `fetchSolcFromGitHub`

Also can we rename the both functions `fetchAndSave...FromGitHub`. While reviewing it was not clear to me if this returns the executable or saves the file to the disk.

Reply:
I thought about it but the check to make sure that the compiler is working is different between the solcjs and the others. Maybe we can just group the download part. Or pass as a callback the check of the working compiler.

Reply:
Do we need to validate it here? The reason we were validating was to check if the binary is executable,  i.e. if you have the Linux solc on a Mac, it will fall back to `solc-js`. I can't think of any case we'd need to "validate" the `solc-js`

Reply:
Anyways, my suggestion would be:
- Have a shared function to `fetchAndSave`
- Don't validate inside but validate in `getSolc` after `fetchAndSave`
- Don't validate `solc-js`, unless we have a reason to do so. If so, validate again in `getSolcJs`.